### ISSUE TITLE
Stable diffusion prompt based on character. 

### ISSUE THREAD

**Issue by Ktrcoyote** [SillyTavern/SillyTavern#612]
**Have you searched for similar [requests](https://github.com/SillyTavern/SillyTavern/issues?q=)?**
Yes

**Is your feature request related to a problem? Please describe.**

SD generation isn't very accurate. Best results come from fitting the prompt prefix individual the character, but generating an image on a second character will still use the same prompt. I think its trying to get character description from the character cards but its not too accurate. I can't tell. 

**Describe the solution you'd like**

user-imputed  prompt prefixes for Stable diffusion based on the character, allowing us to design each person using specific descriptions. we could further increase accuracy with lora's and textual-inversion.

**Comment by Woisek** [comment]
It seems there is no interest in that. I made a similar request and it wasn't even answered ...

**Comment by 7erminalVelociraptor** [comment]
I think this could be helpful. Have a text area in advanced character description that is for physical description only, and have that replace all the other char description in the prompt that get's sent to SD.

Because why does SD need hundreds of tokens explaining the {{char}}'s history with user, their personality, kinks and favorite movies? It easily leads to SD model getting confused and at best it gets ignored and is just wasted promot processing.

**Comment by Woisek** [comment]
@7erminalVelociraptor
It doesn't have to be that complicated or extensive. Like in my already posted feature request, it would be sufficient to have the possibility to edit the prompt before sending it to SD. So instead of clicking and let ST send whatever it wants, having a popup text input for editing the prompt it wants to send and then manual send it after the edit is made.
But like I wrote, the idea is not even commented.

**Comment by 7erminalVelociraptor** [comment]
> @7erminalVelociraptor It doesn't have to be that complicated or extensive. Like in my already posted feature request, it would be sufficient to have the possibility to edit the prompt before sending it to SD. So instead of clicking and let ST send whatever it wants, having a popup text input for editing the prompt it wants to send and then manual send it after the edit is made. But like I wrote, the idea is not even commented.

afaik writing a prompt by hand is already possible. Don't worry about devs not responding btw, because look around on the fid and see most suggestions not having responses. That is just the nature of having only few devs and a lot of daily bug reports and trying to get poe working again every day.

**Comment by Woisek** [comment]
> afaik writing a prompt by hand is already possible.

Yes, I'm aware of the /sd command. But this breaks the chat, because it's a "response" from the bot and not from the system like when clicking on "Generate Image".

@Cohee1207 

>Prompt refinement mode will be added in the future.

Will it be like I suggested or something different? It would be awesome if we could edit the prompt before it gets sent. 👍

**Comment by Woisek** [comment]
> @Woisek The difference is that the slash command/wand menu generates its own prompt, and the brush button uses a specified message as a base, i.e. inline mode.

Yes, that's exactly what I mentioned in the feature request. The brush button would highly benefit (IMHO) from the possibility to refine the prompt before sending it to SD.

> Also refine mode pushed to dev.

So, this feature will come ... ? 😳

**Comment by Ktrcoyote** [comment]
Sorry for the ping, @Cohee1207 , but I believe my original idea was lost in the conversation. While @Woisek 's feature is useful, and it would actually be a feature I'd like to eventually see, just hear me out: 

Allowing the user to input a description for each character and have that prefix injected into the the SD prompt automatically would be the smoothest option, but what's far more interesting is the idea of taking it a step further:  

**Just imagine the potential if the SD character prompt was also integrated into the character cards.**  It'd give creators further control of the people, personalities, and scenarios they create and share, allowing them to indirectly illustrate what that character should look like in SD generation. A particularly determined user could even include a specifically trained LoRa with their card to further solidify the character's look.

Seriously, think of the awesome difference it would make if you could add something like
SDPrompt("An illustration of Raven from Teen Titans <Lora:RavenTT:.5>") to the Raven character profile 
and 
SDPrompt("An illustration of Robin from Teen Titans <Lora:RobinTT:.5>") to Robin's character profile.   
then have SillyTavern automatically inject each prompt prefix depending on which character is talking.

**Comment by 7erminalVelociraptor** [comment]
> So, this feature will come ... ? 😳

It means the feature is already in (with the dev branch), and it will merge into the main branch later

**Comment by Woisek** [comment]
> It means the feature is already in (with the dev branch), and it will merge into the main branch later

Oh, wow, what a great news! 🎉🎉🎉🎉 Many thanks for that! Can't hardly wait to try it out. 😄

**Comment by Woisek** [comment]
> > So, this feature will come ... ? 😳
> 
> It means the feature is already in (with the dev branch), and it will merge into the main branch later

Sorry to ask, but ... where can I find this "refine mode"? Or how can we now edit the prompt from the brush icon? 🤔

**Comment by Cohee1207** [comment]
This was released to stable branch.
